jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -113,24 +113,47 @@ ecma_builtin_array_prototype_object_concat (ecma_value_t this_arg, /**< this arg
   ecma_completion_value_t new_array = ecma_op_create_array_object (0, 0, false);
   ecma_object_t *new_array_p = ecma_get_object_from_completion_value (new_array);
 
+  /* 5.b */
+  if (ecma_object_get_class_name (obj_p) == LIT_MAGIC_STRING_ARRAY_UL)
+  {
+    /* 5.b.iii */
+    for (uint32_t index = 0;
+         index < len && ecma_is_completion_value_empty (ret_value);
+         index++, new_array_index++)
+    {
+      ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
+      /* 5.b.iii.3.a */
+      ECMA_TRY_CATCH (get_value, ecma_op_object_get (obj_p, index_string_p), ret_value);
 
-  for (uint32_t index = 0;
-       index < len && ecma_is_completion_value_empty (ret_value);
-       index++, new_array_index++)
+      /* Using [[Put]] is equvalent to [[DefineOwnProperty]] in this case, so we use it for simplicity. */
+      ecma_completion_value_t put_comp = ecma_op_object_put (new_array_p,
+                                                             index_string_p,
+                                                             get_value,
+                                                             false);
+      JERRY_ASSERT (ecma_is_completion_value_normal (put_comp));
+      ecma_free_completion_value (put_comp);
```

Comment:
As `is_throw` argument of `ecma_op_object_put` is `false`, exceptions can't be thrown and only possible return values in the case are "simple true" or "simple false".
So, we can replace this with corresponding assertion and remove `ecma_free_completion_value` call.